bevy_lint/src/lints/missing_reflect.rs

```diff
@@ -136,7 +136,7 @@ impl TraitType {
     fn from_local_crate<'tcx>(
         tcx: TyCtxt<'tcx>,
         trait_path: &[&str],
-    ) -> impl Iterator<Item = Self> + 'tcx {
+    ) -> impl Iterator<Item = Self> + use<'tcx> {
```

Comment:
question: Can we now do something like this?

```suggestion
    ) -> impl Iterator<Item = Self> {
```

Honestly, I don't fully understand [how the new capturing rules work / what it allows us to do](https://doc.rust-lang.org/nightly/edition-guide/rust-2024/rpit-lifetime-capture.html), but might be worth a try to simplify this.

Reply:
We _could_, but it wouldn't be correct. If we remove the `use<..>` bound, the compiler assumes that the returned `impl Iterator` type captures all lifetimes in function header.

To put it in other words, your suggestion would make the compiler see the function as this:

```rust
fn from_local_crate<'tcx, 'a, 'b>(
    tcx: TyCtxt<'tcx>,
    trait_path: &'a [&'b str],
) -> impl Iterator<Item = Self> + use<'tcx, 'a, 'b> {
```

This unfortunately isn't true, because the returned iterator doesn't capture `trait_path`. (You'll see that `trait_path` is only used once, in `def_path_res()`.)

So yes, we could, but it would place incorrect restrictions on how the returned iterator can be used. (Even if in practice is doesn't matter.)